src/aktualizr_secondary/secondary_tcp_server.cc

```diff
@@ -157,15 +172,22 @@ void SecondaryTcpServer::HandleOneConnection(int socket) {
       asn_enc_rval_t encode_result =
           der_encode(&asn_DEF_AKIpUptaneMes, &resp->msg_, Asn1SocketWriteCallback, reinterpret_cast<void *>(&socket));
       if (encode_result.encoded == -1) {
-        return;  // write error
+        return true;  // write error
       }
       optval = 1;
       setsockopt(socket, IPPROTO_TCP, TCP_NODELAY, &optval, sizeof(int));
     } else {
       LOG_DEBUG << "Not sending a response to message " << msg->present();
     }
+
+    if (need_reboot && reboot_after_install_) {
+      exit_reason_ = ExitReason::kRebootNeeded;
+      return false;
```

Comment:
As an option `exit_reason_` can be returned from HandleOneConnection to run() and run() in turn can return the exit status to a caller/client so no class member is required to store an exit reason and no additional call for a client to get this exit reason. But it' really minor and not important just an alternative.